lib/Mojolicious.pm

```diff
@@ -175,6 +175,16 @@ sub plugin {
 
 sub server { $_[0]->plugins->emit_hook(before_server_start => @_[1, 0]) }
 
+sub app_starting {
+  my $self = shift;
+
+  # Ensure we run 'before_app_start' only once per process
+  $self->{_started} //= {};
+  return if exists $self->{_started}->{$$};
+  $self->{_started}->{$$} = undef;
```

Comment:
This should be completely unnecessary.

Reply:
It's not.  See above comment... If I do not do this, then there is a duplicate invocation (as you guessed) with having run() and start() both invoke the event...
Yet if both do not, then the event is not always emitted.

Reply:
It's a workaround, not a solution for the actual problem (which is that morbo initialises the listener in the parent process before the application is ever loaded).